### PR TITLE
feat: Add rule_type parameter to merge_request_approval_rule

### PR DESCRIPTION
**Pull request by petrmvala**
This is working in the API, and is necessary in order to enable
Eligible Approvers.

## Description

#904 

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

File: internal/provider/resource_gitlab_project_approval_rule.go [hunk 1]
```diff
@@ -72,6 +77,7 @@ func resourceGitlabProjectApprovalRuleCreate(ctx context.Context, d *schema.Reso
 	options := gitlab.CreateProjectLevelRuleOptions{
 		Name:               gitlab.String(d.Get("name").(string)),
 		ApprovalsRequired:  gitlab.Int(d.Get("approvals_required").(int)),
+		RuleType:           gitlab.String(d.Get("rule_type").(string)),
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project_approval_rule.go:80]: If an attribute is not set in the config and is optional it's good to also not send it to the API, therefore, I'd do it like this:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/278f2b394ec57caf9e415c2b42d31bdb48b76e78/internal/provider/resource_gitlab_topic.go#L56

I'm aware that it also applies to the attributes already in here. It's never to late to start though and if you want feel free to apply it for the others, too 🌮

File: internal/provider/resource_gitlab_project_approval_rule_test.go [hunk 2]
```diff
@@ -57,12 +58,82 @@ func TestAccGitLabProjectApprovalRule_basic(t *testing.T) {
...
+			{
+				ResourceName:      "gitlab_project_approval_rule.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccGitLabProjectApprovalRule_any_approver(t *testing.T) {
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project_approval_rule_test.go:84]: NIT:

```suggestion
func TestAccGitLabProjectApprovalRule_anyApprover(t *testing.T) {
```

**Review by timofurrer — CHANGES_REQUESTED** [review]
Looks very promising, nice work! I've left a few minor comments.

File: internal/provider/resource_gitlab_project_approval_rule.go [hunk 3]
```diff
@@ -43,6 +43,11 @@ var _ = registerResource("gitlab_project_approval_rule", func() *schema.Resource
 				Type:        schema.TypeInt,
 				Required:    true,
 			},
+			"rule_type": {
+				Description: "The type of rule. `any_approver` is a pre-configured default rule with `approvals_required` at `0`. Other rules are `regular`.",
```
[the remark this answers is not in the document]
Comment (petrmvala) [reply internal/provider/resource_gitlab_project_approval_rule.go:47]: Yes, exactly. I took the docs text from here: https://docs.gitlab.com/14.7/ee/api/merge_request_approvals.html#create-project-level-rule

Thanks for the comments and let me see :)

File: internal/provider/resource_gitlab_project_approval_rule.go (hunk 1 above)
Comment (petrmvala) [reply internal/provider/resource_gitlab_project_approval_rule.go:80]: I would prefer to touch only this one in this scope and then go through the others separately, would that be ok?

Comment (timofurrer) [reply internal/provider/resource_gitlab_project_approval_rule.go:80]: sure thing! You can also leave it and I'll sweep over it afterwards ;)

File: internal/provider/resource_gitlab_project_approval_rule_test.go (hunk 2 above)
Comment (petrmvala) [reply internal/provider/resource_gitlab_project_approval_rule_test.go:84]: Actually, shouldn't it be `_AnyApprover` ?

File: internal/provider/resource_gitlab_project_approval_rule.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (timofurrer) [reply internal/provider/resource_gitlab_project_approval_rule.go:47]: Yes, I saw that, too - docs are sometimes incomplete (🤫 ) that's why I've asked

File: internal/provider/resource_gitlab_project_approval_rule.go [hunk 4]
```diff
@@ -3,17 +3,24 @@ package provider
...
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 	gitlab "github.com/xanzy/go-gitlab"
 )
 
 // https://docs.gitlab.com/ee/api/merge_request_approvals.html#create-project-level-rule
 var errApprovalRuleNotFound = errors.New("approval rule not found")
 
+var validRuleTypeValues = []string{
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project_approval_rule.go:19]: Let's move this in the `registerResource()` call between current line 24 / 25

**Review by timofurrer — APPROVED** [review]
LGTM 🎉

**Comment by timofurrer** [comment]
@DonNicoJs @armsnyder I've merged this because all checks passed and I've assigned the `safe to test` label (this is an EE feature) - However, seems like EE tests were not executed and `main` is now failing :'( Not sure what went wrong here. But I'll try to make `main` pass again ;)

**Pull request merged**